zipkin-lens/src/components/App/App.test.js

```diff
@@ -11,54 +11,61 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-import { shallow } from 'enzyme';
 import React from 'react';
 import { Provider } from 'react-redux';
 import { BrowserRouter, Route } from 'react-router-dom';
 import { ThemeProvider } from '@material-ui/styles';
 import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { createMount, createShallow } from '@material-ui/core/test-utils';
 
 import App from './App';
 import Layout from './Layout';
+import DiscoverPage from '../DiscoverPage';
+import TracePage from '../TracePage';
+import { theme } from '../../colors';
 
 describe('<App />', () => {
-  let wrapper;
+  let mount;
+  let shallow;
 
   beforeEach(() => {
-    wrapper = shallow(<App />);
+    mount = createMount({ strict: true });
+    shallow = createShallow();
   });
 
-  it('should render Layout', () => {
-    const items = wrapper.find(Layout);
-    expect(items.length).toBe(1);
+  afterEach(() => {
+    mount.cleanUp();
   });
 
-  describe('should render router', () => {
-    it('should render BrowserRouter', () => {
-      const items = wrapper.find(BrowserRouter);
-      expect(items.length).toBe(1);
-    });
+  it('should update document title', () => {
+    mount(<App />);
```

Comment:
for my learning.. mount() seems to be documented for things that need access to whole dom. Can you briefly mention why we need whole dom for this test? I guess shallow doesn't set the title? https://material-ui.com/guides/testing/#full-dom-rendering